src/main/java/seedu/address/model/task/Priority.java

```diff
@@ -51,14 +62,23 @@ public int getEnumPriority(){
     public Priority(String priority) throws IllegalValueException{
         assert priority != null;
         priority = priority.trim();
+        
         String upperPriority = priority.toUpperCase();
+        
+        listOfPriorities = new HashMap<String, String>();
+        getListOfDifferentAliases();
```

Comment:
Not a good idea to be calling this in the constructor, this causes the values to be cleared added repeatedly every time a new Priority is created.

Maybe try initializing it just once on init instead? You can view an example in TaskPropertyMapping under ui.